kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -7015,11 +7015,42 @@ protected void handleAccessResource(HttpServletRequest req, HttpServletResponse
 					return; 
 				}
 
+				// If there is a direct link to the asset, no sense streaming it.
+				// Send the asset directly to the load-balancer or to the client
+				URI directLinkUri = m_storage.getDirectLink(resource);
```

Comment:
The only thing that comes to mind is the content filter stuff having set a conflicting header. It's safe to overwrite header values, so this is a long shot.

It looks like the wrapper (resource and stream) handles things gracefully. If you have some repro steps, I can fire up a debugger.